src/Kernel/Process.class.st

```diff
@@ -324,18 +368,18 @@ Process >> isSuspended [
 
 { #category : #testing }
 Process >> isTerminated [
+
+	"Answer if the receiver is terminated, i.e. if the receiver is not active and 
+	one of the following conditions is met:
+	(1) the receiver is a defunct process (suspendedContext = nil or pc = nil)
+	(2) the receiver is suspended in the endProcess method (We cannot use pragmas... pragmas require the compiler, it generates an evil dependency.
+	It also is not available in minimal images)"
+
 	self isActiveProcess ifTrue: [ ^ false ].
 	
-	^ suspendedContext isNil or: [
-		"If the suspendedContext is the bottomContext it is the block in Process>>newProcess.
-		If so, and the pc is greater than the startpc, the block has already sent and returned
-		from value and there is nothing more to do."
-		suspendedContext isBottomContext and: [
-			suspendedContext isDead or: [
-				"The pc of the suspendedContext is set to nil in #terminate explicitly.
-				Leaving this line in for safety."
-  	  			suspendedContext pc > suspendedContext startpc ] ] ]
-
+	^suspendedContext isNil or: [
+		suspendedContext isDead or: [
+			(suspendedContext method == (self class >> #endProcess))]]
```

Comment:
See my previous comments. #isDead already incorporates this logic but it needs to be adjusted for #endProcess method